File: CSharp/TensionsBetweenDifferentOOPrinciples/TensionBetweenDRYandLowCoupling/After/UniblabRobot.cs
using System;

namespace TensionBetweenDRYandLowCoupling.After
{
    public class UniblabRobot : Robot
    {
		// UniblabRobot class represents a robot that accepts commands to move in four directions.
		// UniblabRobot class follows the DRY principle now because code in common with RosieRobot is in Robot class.
		// UniblabRobot class has one more dependencies to Robot class, so coupling is higher.
		// Sometimes this coupling in called 'subclass coupling' or, more in general, a 'dependency'.

		private Tuple<int, int> _currentPosition = new Tuple<int, int>(5, 5);

        public void MoveOneNorthEast()
        {
            _currentPosition = Add(_currentPosition, Add(North, East));
        }

        public void MoveOneNorthWest()
        {
            _currentPosition = Add(_currentPosition, Add(North, West));
        }

        public void MoveOneSouthEast()
        {
            _currentPosition = Add(_currentPosition, Add(South, East));
        }

        public void MoveOneSouthWest()
        {
            _currentPosition = Add(_currentPosition, Add(South, West));
        }
 
    }
}

File: javascript/TensionsBetweenDifferentOOPrinciples/TensionBetweenDRYandLowCoupling/After/uniblab_robot.js
var Robot = require("./robot.js");

// UniblabRobot class represents a robot that accept commands to move in four directions.
// UniblabRobot class follows the DRY principle now because code in common with RosieRobot is in Robot class.
// UniblabRobot class has one more dependency to Robot class so coupling is higher.
var UniblabRobot = function () {
  this.currentPosition = [5, 5];
};

UniblabRobot.prototype = Object.create(Robot.prototype);

UniblabRobot.prototype.moveOneNorthEast = function () {
  this.currentPosition = this.add(this.currentPosition, this.add(this.north, this.east));
};

UniblabRobot.prototype.moveOneNorthWest = function () {
  this.currentPosition = this.add(this.currentPosition, this.add(this.north, this.west));
};

UniblabRobot.prototype.moveOneSouthEast = function () {
  this.currentPosition = this.add(this.currentPosition, this.add(this.south, this.east));
};

UniblabRobot.prototype.moveOneSouthWest = function () {
  this.currentPosition = this.add(this.currentPosition, this.add(this.south, this.west));
};

module.exports = UniblabRobot;
